### PR TITLE
Add log4j and basic logging

### LINKED ISSUE #75: [Vulnerability] Improve logging
Perform appropriate logging using the log4j Java framework to record sensitive operations, such as logins, and access to/modification of sensitive information.

### PR DESCRIPTION
**Pull request by AmplifiedHuman**
- added info and error logs to report application state

closes #75

### CODE REVIEW

File: backend/src/main/java/com/vax/warden/controller/ForumController.java [hunk 1]
```diff
@@ -19,24 +21,28 @@
...
 public class ForumController {
     private final ForumService forumService;
+    private static final Logger logger = LogManager.getLogger(ForumController.class);
 
     @PostMapping("/post")
     @ResponseStatus(HttpStatus.CREATED)
     public Post post(@Valid @RequestBody Post post, Authentication authentication) {
         String email = (String) authentication.getPrincipal();
+        logger.info("Creating new post: " + post.getId());
         return forumService.createPost(post, email);
```
Comment (rajitbanerjee) [inline backend/src/main/java/com/vax/warden/controller/ForumController.java:31]:
```suggestion
        Post newPost = forumService.createPost(post, email);
        logger.info("Creating new post: " + newPost.getId());
        return newPost;
```

File: backend/src/main/java/com/vax/warden/controller/ForumController.java [hunk 2]
```diff
@@ -19,24 +21,28 @@
...
     }
 
     @PostMapping("/reply")
     @ResponseStatus(HttpStatus.CREATED)
     public Post reply(@Valid @RequestBody Post post, Authentication authentication) {
         String email = (String) authentication.getPrincipal();
+        logger.info("Creating new reply for post: " + post.getId());
```
Comment (rajitbanerjee) [inline backend/src/main/java/com/vax/warden/controller/ForumController.java:38]:
```suggestion
        logger.info("Creating new reply for post: " + post.getReplyToPostId());
```

**Review by rajitbanerjee — CHANGES_REQUESTED** [review]
Mostly LGTM, added 3 more commits:
- Removing flyway from pom.xml - was supposed to be removed in a previous PR, but possibly messed up the merge conflict resolution
- Ran mvn clean install for auto format to sort imports
- Set jpa.show-sql = false to avoid log pollution since log4j already takes care of it

Requested changes:
- When new post is created, it needs to call the service function or else the id is null
- The post being replied to is fetched using replyToPostId()

**Comment by AmplifiedHuman** [comment]
Both changes makes sense, committed suggested change, good to go.

**Review by rajitbanerjee — APPROVED** [review]

**Pull request merged**